### ISSUE TITLE
Wrongly reported Missing backslash-space between literal and text

### ISSUE THREAD

**Issue by matilda-me** [sphinx-contrib/sphinx-lint#15]
In pandas doc, the following is valid rst:

```rst
* A pandas class (in the form ``:class:`pandas.Series```)
* A pandas method (in the form ``:meth:`pandas.Series.sum```)
* A pandas function (in the form ``:func:`pandas.to_datetime```)
```

but sphinx-lint reports:

```test
[2] pandas/doc/source/development/contributing_docstring.rst:111: Missing backslash-space between literal and text (column 53).
[2] pandas/doc/source/development/contributing_docstring.rst:112: Missing backslash-space between literal and text (column 57).
[2] pandas/doc/source/development/contributing_docstring.rst:113: Missing backslash-space between literal and text (column 60).
```
